controllers/api/lib/application-expiry.js

```diff
@@ -0,0 +1,124 @@
+        throw new Error(
+            'Missing environment variable APPLICATION_EXPIRY_EMAIL'
+        );
+    }
+
+    const getLabelForEmailType = type => {
+        const email = Object.values(EXPIRY_EMAIL_REMINDERS).find(
+            email => email.key === type
+        );
+        return email
+            ? email.label[locale]
+            : EXPIRY_EMAIL_REMINDERS.ONE_DAY.label[locale];
+    };
+
+    return await Promise.all(
+        emailQueue.map(async emailToSend => {
+            let returnObj = { emailSent: false, dbUpdated: false };
+
+            const mailParams = {
+                name: 'application_expiry_afa',
+                sendTo: appData.isNotProduction
+                    ? process.env.APPLICATION_EXPIRY_EMAIL
+                    : emailToSend.PendingApplication.user.username,
+                subject: `You have ${getLabelForEmailType(
+                    emailToSend.emailType
+                )} to finish your application`
+            };
+
+            const getAppData = field =>
+                get(emailToSend.PendingApplication, `applicationData.${field}`);
+
+            const emailStatus = await sendHtmlEmail(
+                {
+                    template: path.resolve(__dirname, './expiry-email.njk'),
+                    templateData: {
+                        timeToFinishApp: getLabelForEmailType(
+                            emailToSend.emailType
+                        ),
+                        projectName: getAppData('projectName'),
+                        countryPhoneNumber: getPhoneFor(
+                            getAppData('projectCountry')
+                        ),
+                        countryEmail: getEmailFor(getAppData('projectCountry')),
+                        application: emailToSend.PendingApplication
+                    }
+                },
+                mailParams
+            );
+
+            if (emailStatus.response || appData.isTestServer) {
+                returnObj.emailSent = true;
+
+                const dbStatus = (await ApplicationEmailQueue.updateStatusToSent(
+                    emailToSend.id
+                ))[0];
+
+                if (dbStatus === 1) {
+                    returnObj.dbUpdated = true;
+                }
+            }
+            return returnObj;
+        })
+    );
+};
+
+/**
+ * Handle Expired Applications:
+ * First deletes the foreignKey references
+ * Then, deletes all the expired applications (no emails sent)
+ * DELETE promise returns number of records affected directly
+ * returns truthy only if no. of del records = no. of expired applications
+ */
+const deleteExpiredApplications = async expiredApplications => {
+    try {
+        logger.info('Handling expired applications');
+        const applicationIds = expiredApplications.map(
+            application => application.id
+        );
+
+        const dbStatus = await PendingApplication.deleteApplications(
+            applicationIds
```

Comment:
Guess we will also have to delete the emailQueue records as they serve no purpose without the pending application (or does the relation take care of it ?).

Reply:
We can keep them around for a while, eg. 3 months after they're sent – it helps for auditing purposes, eg. if someone complains that we deleted their application without warning them, we can look up the record to verify that we did email them.